### ISSUE TITLE
Possible Issue with keras estimator's get_activations() function when framework = True

### ISSUE THREAD

**Issue by keykholt** [Trusted-AI/adversarial-robustness-toolbox#1494]
**Describe the bug**
When the framework flag is true, `get_activations()` attempts to create a placeholder object and a new output tensor to be returned. However, the functionality does not work when the target layer requires a certain input shape. Layers such as Dropout, Flatten, and Maxpooling do not have an issue, but layers such as Conv2D and Dense will have the issue.

As an additional issue, even when function returns a placeholder and tensor, I don't think the returned values will be useful when evaluated as they do not accurately represent the intermediate activation of the model, but rather the activation of the input directly evaluated by the current layer.

**To Reproduce**
Given a keras estimator and a batch of inputs x, for the first issue described:
1. Identify an intermediate layer that is a Dense or Conv layer. Let us denote this layer as `feature_layer`
2. Call `estimator.get_activations(x, feature_layer, 1)`. This should work and return without error.
3. Call `estimator.get_activations(x, feature_layer, 1, framework=True)`. This should break and thrown an error, which states that the final axis shape is different from the expected_value

Given a keras estimator and a batch of inputs x, for the second issue described:
1. Identify an intermediate layer that is a Maxpooling, Flatten, or Dropout layer, preferably near the end of the model. Let us denote this layer as `feature_layer`
2. Call `ret_val_1 = estimator.get_activations(x, feature_layer, 1)` and save the value.
3. Call `placeholder, output_tensor = estimator.get_activations(x, feature_layer, 1, framework=True)`. This should work.
4. Import the keras backend (e.g. `import tensorflow.keras.backend as k`)
5. Call `temp = k.function([placeholder, k.learning_phase()], [output_tensor])`
6. Evaluate the output tensor: `ret_val_2 = temp(x)
7. Compare `ret_val_1` and `ret_val_2`. They will be different both in value and probably shape.

**Expected behavior**
For the first issue, we would expect `get_activations` to work. For the second issue, we would expect `ret_val_1 == ret_val_2`.

**System information (please complete the following information):**
 - Unbuntu 18.04 
 - Python 3.8
 - ART 1.9.1
 - Keras 2.6

**Comment by beat-buesser** [comment]
Hi @keykholt Thank you very much for noticing and reporting the issue in detail!
